cmd/influx/task.go

```diff
@@ -3,342 +3,458 @@ package main
 			return err
 		}
 
-		task, err := s.FindTaskByID(context.Background(), *id)
+		task, err := tskSvc.FindTaskByID(context.Background(), *id)
 		if err != nil {
 			return err
 		}
 
 		tasks = append(tasks, task)
 	} else {
-		tasks, _, err = s.FindTasks(context.Background(), filter)
+		tasks, _, err = tskSvc.FindTasks(context.Background(), filter)
 		if err != nil {
 			return err
 		}
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			tasks:       tasks,
-		},
-	)
+	return b.printTasks(taskPrintOpts{tasks: tasks})
+}
+
+type taskRerunFailedFlags struct {
+	before      string
+	after       string
+	executeRuns bool
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedCmd() *cobra.Command {
+	cmd := b.opts.newCmd("retry-failed", b.taskRetryFailedF, true)
+	cmd.Short = "Retry failed runs"
+	cmd.Aliases = []string{"rtf"}
+
+	b.org.register(b.opts.viper, cmd, false)
+	b.globalFlags.registerFlags(b.opts.viper, cmd)
+	registerPrintOptions(b.opts.viper, cmd, &b.taskPrintFlags.hideHeaders, &b.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&b.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.before, "before", "", "runs before this time")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.after, "after", "", "runs after this time")
+	cmd.Flags().BoolVar(&b.taskRerunFailedFlags.executeRuns, "run", false,
+		"actually execute failed runs; defaults false")
+
+	return cmd
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedF(*cobra.Command, []string) error {
+	if err := b.org.validOrgFlags(&flags); err != nil {
+		return err
+	}
+	tskSvc, _, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+
+	var failedRuns []*influxdb.Run
+	if b.taskID == "" {
+		failedRuns, err = b.getFailedRunsForOrg()
+	} else {
+		failedRuns, err = b.getFailedRunsForTaskID()
+	}
+	if err != nil {
+		return err
+	}
+
+	for _, run := range failedRuns {
+		if b.taskRerunFailedFlags.executeRuns {
+			newRun, err := tskSvc.RetryRun(context.Background(), run.TaskID, run.ID)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Retry for task %s's run %s queued as run %s.\n", run.TaskID, run.ID, newRun.ID)
+		} else {
+			fmt.Printf("Would retry for %s run for Task %s.\n", run.ID, run.TaskID)
+		}
+
+	}
+	return nil
+
+}
+
+func (b *cmdTaskBuilder) getFailedRunsForTaskID() ([]*influxdb.Run, error) {
+	// use RunFilter to search for failed runs
+	tskSvc, _, err := b.svcFn()
+	if err != nil {
+		return nil, err
+	}
+	runFilter := influxdb.RunFilter{}
+	id, err := influxdb.IDFromString(b.taskID)
+	if err != nil {
+		return nil, err
+	}
+	runFilter.Task = *id
+	runFilter.BeforeTime = b.taskRerunFailedFlags.before
+	runFilter.AfterTime = b.taskRerunFailedFlags.after
+	allRuns, _, err := tskSvc.FindRuns(context.Background(), runFilter)
+	if err != nil {
+		return nil, err
+	}
+	var allFailedRuns []*influxdb.Run
+	for _, run := range allRuns {
+		if run.Status == "failed" {
```

Comment:
Could we move this filter into the `runFilter` construction, so we only pull relevant things from the server?

Reply:
unfortunately, none of the filters (both runs and task) do not support filtering by status.